cmd/controller/main.go

```diff
@@ -134,9 +134,9 @@ func main() {
 			brokerInformer,
 		),
 	}
-	if len(controllers) != numControllers {
-		logger.Fatalf("Number of controllers and QPS settings mismatch: %d != %d", len(controllers), numControllers)
-	}
+	controllers := controllersArray[:]
+	// compile-time assert numControllers == len(controllersArray)
```

Comment:
This would be clearer as a complete sentence. I suggest something like

```go
// This line asserts at compile time that the length of controllersArray is equal to numControllers.
```